contracts/FundManagement.sol

```diff
@@ -0,0 +1,167 @@
+ */
+contract FundManagement is Ownable {
+    struct Fund {
+        uint256 totalAmount; // Total amount of tokens assigned to the manager
+        address asset;
+        uint256 requestedWithdrawTime; // Timestamp which the manager may withdraw the pendingWithdrawAmount
+        uint256 pendingWithdrawAmount; // Total amount of tokens the manager is pending to withdraw
+    }
+
+    using SafeERC20 for IERC20;
+
+    uint256 public requestWindow = 2 days;
+    mapping(address => mapping(uint256 => Fund)) public funds;
+    mapping(address => uint256) public numberOfFunds;
+    mapping(address => uint256) public locked;
+
+    event CreateFund(address indexed manager, address indexed asset, uint256 amount);
+    event RequestFunds(address indexed to, uint256 amount);
+    event Claim(address indexed to, uint256 amount);
+    event Clawback(address indexed account, uint256 fundNumber);
+
+    constructor() {}
+
+    function requestFunds(uint256 fundNumber, uint256 amount) external {
+        require(
+            fundNumber >= 0 && fundNumber < numberOfFunds[msg.sender],
+            "The input fund number does not exist"
+        );
+        Fund storage fund = funds[msg.sender][fundNumber];
+        uint256 totalWithdrawableAmount = fund.pendingWithdrawAmount + amount;
+        require(
+            totalWithdrawableAmount <= fund.totalAmount, 
+            "Requested amount plus withdrawable amount greater than total allocated funds"
+        );
+        
+        fund.requestedWithdrawTime = block.timestamp + requestWindow;
+        fund.pendingWithdrawAmount = totalWithdrawableAmount;
+
+        emit RequestFunds(msg.sender, amount);
+    }
+
+    function claim(uint256 fundNumber) external {
+        require(
+            fundNumber >= 0 && fundNumber < numberOfFunds[msg.sender],
+            "The input fund number does not exist"
+        );
+        Fund storage fund = funds[msg.sender][fundNumber];
+        uint256 pendingAmount = fund.pendingWithdrawAmount;
+
+        require(
+            pendingAmount > 0,
+            "No withdrawable funds"
+        );
+        require(
+            pendingAmount <= fund.totalAmount, 
+            "Withdrawable amount greater than total allocated funds"
+        );
+        require(
+            block.timestamp >= fund.requestedWithdrawTime,
+            "Your funds are not withdrawable yet"
+        );
+
+        locked[fund.asset] = locked[fund.asset] - pendingAmount;
+        fund.totalAmount = fund.totalAmount - pendingAmount;
+        fund.pendingWithdrawAmount = 0;
+
+        IERC20(fund.asset).safeTransfer(msg.sender, pendingAmount);
+
+        emit Claim(msg.sender, pendingAmount);
+    }
+
+    /**
+     * @notice allows owner to change the duration of time a fund manager needs to wait to withdraw funds they request.
+     * @param duration of time in seconds.
+     */
+    function setRequestWindow(uint256 duration) external onlyOwner {
+        requestWindow = duration;
```

Comment:
I missed this before, but you should emit an event for a state variable change.

Reply:
Also, setters are typically at the bottom of contracts (if modifiers are in the contract, above them).